tests/slow_tests/test_benchmarks.py

```diff
@@ -1,37 +1,94 @@
-from flow.benchmarks.baselines.bottleneck2 import bottleneck2_baseline
-from flow.benchmarks.baselines.figureeight012 import figure_eight_baseline
-from flow.benchmarks.baselines.grid0 import grid0_baseline
-from flow.benchmarks.baselines.grid1 import grid1_baseline
-from flow.benchmarks.baselines.merge012 import merge_baseline
+import ray
+import ray.rllib.agents.ppo as ppo
+
+from ray.rllib.agents.agent import get_agent_class
+from ray.tune.registry import register_env
+
+from flow.utils.registry import make_create_env
+from flow.utils.rllib import FlowParamsEncoder
+
+
+import flow.benchmarks.bottleneck0 as bottleneck0
+import flow.benchmarks.bottleneck1 as bottleneck1
+import flow.benchmarks.bottleneck2 as bottleneck2
+import flow.benchmarks.figureeight0 as figureeight0
+import flow.benchmarks.figureeight1 as figureeight1
+import flow.benchmarks.figureeight2 as figureeight2
+import flow.benchmarks.grid0 as grid0
+import flow.benchmarks.grid1 as grid1
+import flow.benchmarks.merge0 as merge0
+import flow.benchmarks.merge1 as merge1
+import flow.benchmarks.merge2 as merge2
+
+N_CPUS = 1
+ray.init(num_cpus=N_CPUS, redirect_output=True)
 
 os.environ["TEST_FLAG"] = "True"
 
 
-class TestBaselines(unittest.TestCase):
+class TestBenchmarks(unittest.TestCase):
+
     """
     Tests that the baselines in the benchmarks folder are running and
     returning expected values (i.e. values that match those in the CoRL paper
     reported on the website, or other).
     """
 
+    """
+    self.env, self.scenario = setup_bottlenecks()
+        self.exp = SumoExperiment(self.env, self.scenario)
+        self.exp.run(5, 50)
+    """
+
+    def ray_runner(self, num_runs, flow_params, version):
+        alg_run = "PPO"
+        HORIZON = 10
+        N_ROLLOUTS = 1
+
+        agent_cls = get_agent_class(alg_run)
+        config = agent_cls._default_config.copy()
+        config["num_workers"] = 1
+        config["timesteps_per_batch"] = HORIZON * N_ROLLOUTS
+        config["gamma"] = 0.999  # discount rate
+        config["model"].update({"fcnet_hiddens": [100, 50, 25]})
+        config["use_gae"] = True
+        config["lambda"] = 0.97
+        config["kl_target"] = 0.02
+        config["num_sgd_iter"] = 1
+        config["horizon"] = HORIZON
+        config["observation_filter"] = "NoFilter"
+
+        # save the flow params for replay
+        flow_json = json.dumps(
+            flow_params, cls=FlowParamsEncoder, sort_keys=True, indent=4)
+        config['env_config']['flow_params'] = flow_json
+        config['env_config']['run'] = alg_run
+
+        create_env, env_name = make_create_env(params=flow_params,
+                                               version=version)
+
+        # Register as rllib env
+        register_env(env_name, create_env)
+
+        alg = ppo.PPOAgent(
+            env=env_name, config=config)
+
+        for i in range(num_runs):
+            alg.train()
+            checkpoint_path = alg.save()
```

Comment:
Can you remove the checkpoint files afterwards?

Reply:
Addressed!